sycl/include/CL/sycl/detail/backend_traits_level_zero.hpp

```diff
@@ -130,11 +134,61 @@ template <> struct BackendReturn<backend::ext_oneapi_level_zero, event> {
   using type = ze_event_handle_t;
 };
 
+struct OptionalDevice {
```

Comment:
Why do you need this new class? Why can't the "optional" nature of the device be expressed with just allowing `nullptr` in the `sycl::device` ?

Reply:
@gmlueck and I discussed this in https://github.com/intel/llvm/pull/6148#discussion_r872356058. I believe we could do that, though my primary concern is that we expose a "fake" device that the user could end up passing around, potentially causing future headaches.

Reply:
I see. I actually mistakenly assumed that "device" was a pointer. So I see why you need this wrapper class.